### PR TITLE
Fixes #766 Show notifications when network status of device changes.

### LINKED ISSUE #766: Show notifications when network status of device changes.
**Summary:** 

Whenever the user goes offline or online,a snackBar should give notification to the User.

**Screen-shots:** 
whatsapp image 2018-02-16 at 00 06 57-2
whatsapp image 2018-02-16 at 00 06 57

### PR DESCRIPTION
**Pull request by Siddharth-Singhs**
## Description

Added the internet Status

 ## Screen-shots, if any
whatsapp image 2018-02-18 at 15 35 54-2
whatsapp image 2018-02-18 at 15 35 54

## Related issues and discussion
#766

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java [hunk 1]
```diff
@@ -541,4 +570,84 @@ public void moveToBarcodeEntry(){
...
+    private void setSnackbarMessage(String status,boolean showBar) {
+        String internetStatus="";
+        if(status.equalsIgnoreCase("Wifi enabled")||status.equalsIgnoreCase("Mobile data enabled")){
+            internetStatus="Internet Connected";
+        }else {
+            internetStatus="Lost Internet Connection";
+        }
+        snackbar = Snackbar
+                .make(linearLayout, internetStatus, Snackbar.LENGTH_LONG)
+                .setAction("X", new View.OnClickListener() {
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java]: Can you change this to an action button that leads the user to the network settings of the device instead of cancelling the snackbar?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java [hunk 2]
```diff
@@ -541,4 +570,84 @@ public void moveToBarcodeEntry(){
...
+                    @Override
+                    public void onClick(View view) {
+                        snackbar.dismiss();
+                    }
+                });
+        // Changing message text color
+        snackbar.setActionTextColor(Color.WHITE);
+        // Changing action button text color
+        View sbView = snackbar.getView();
+        TextView textView = (TextView) sbView.findViewById(android.support.design.R.id.snackbar_text);
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java]: You do not need to get the textview for a Snackbar, there is a method already built into this.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java [hunk 3]
```diff
@@ -541,4 +570,84 @@ public void moveToBarcodeEntry(){
...
+     */
+    public BroadcastReceiver broadcastReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            String status = getConnectivityStatusString(context);
+            setSnackbarMessage(status,false);
+        }
+    };
+
+    public static int getConnectivityStatus(Context context) {
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java]: Please add this method to the Utils class to ensure that this code is reusable.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java [hunk 4]
```diff
@@ -541,4 +570,84 @@ public void moveToBarcodeEntry(){
...
+        }
+        return status;
+    }
+    private void setSnackbarMessage(String status,boolean showBar) {
+        String internetStatus="";
+        if(status.equalsIgnoreCase("Wifi enabled")||status.equalsIgnoreCase("Mobile data enabled")){
+            internetStatus="Internet Connected";
+        }else {
+            internetStatus="Lost Internet Connection";
+        }
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java]: Can you rephrase these as:
- Network Status: disconnected
- Network Status: connected

Comment (policorn) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java]: For the internationalization sake, I'd like to ask for these strings be references to resources

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java [hunk 5]
```diff
@@ -541,4 +570,84 @@ public void moveToBarcodeEntry(){
...
+                        snackbar.dismiss();
+                    }
+                });
+        // Changing message text color
+        snackbar.setActionTextColor(Color.WHITE);
+        // Changing action button text color
+        View sbView = snackbar.getView();
+        TextView textView = (TextView) sbView.findViewById(android.support.design.R.id.snackbar_text);
+        textView.setTextColor(Color.WHITE);
+        if(internetStatus.equalsIgnoreCase("Lost Internet Connection")){
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java]: i18n string here please 😄
